### DISCUSSION TITLE
Python script to automatically create multiple images of a rotating knob

### DISCUSSION THREAD

**Discussion by dervur** [RomanKubiak/ctrlr#663] [Show and tell]
Hi everybody, 

Maybe some of you already have scripts to do this, but on my second day of Ctrlr it became clear to me that if I wanted to use an image modulator for a rotating knob, creating 128 images of a knob at different position would require an insane a mount of time. I thought there had to be an easier way to do it... So I thought about python and wrote a simple script for it.

Right now the script can rotate and saturate the image as it goes, I had this for a knob I wanted to change colour as it was rotated.
Change the values as you please to achieve the result you want. The name of the exported file will also have the frequency and sizes needed for Ctrlr to correctly visualize the figure (just in case you forget these in a second moment).

As an example this was the knob I wanted to change colour to as it rotated. Set at minimum position at 8 o'clock-ish
![knob pink min](https://github.com/RomanKubiak/ctrlr/assets/77355362/854ebc19-597d-4186-a6b9-4c6310fd85e6)

And these are some position of this image along the line:
![image](https://github.com/RomanKubiak/ctrlr/assets/77355362/a6d248fe-a1c7-4bff-a588-2256ccd2604a)
![image](https://github.com/RomanKubiak/ctrlr/assets/77355362/df43a938-0436-4eb5-b8b5-cb5d843f3a3b)
![image](https://github.com/RomanKubiak/ctrlr/assets/77355362/dee251de-e57f-406c-960f-09a11e8b0b94)

You will need the PILLOW library: https://stackoverflow.com/questions/68439152/how-to-install-pil-with-pip

Here is the script:
[tile knob for Ctrlr.zip](https://github.com/user-attachments/files/16096755/tile.knob.for.Ctrlr.zip)

I don't know if other ways to do these are available, but this is the one I implemented and I wanted to share with you as it might be useful to some, especially people approaching Ctrlr. There are some comments to help you in choosing how to select some values.
It is quite basic but feel free to expand it if you want with more features and feel free to post them here so others can implement them as well :)

**Comment by damiensellier** [comment]
i think we all use knobman for that particular purpose.
you can use layers and rotate your marker only above the fixed background knob.

https://www.g200kg.com/jp/software/knobman.html#nativeknobman

NOTE : For retina/HiDPI purpose, always use x2 or x4 size of the knob to avoid jagged lines when scaling up.
A 60x60px knob should have a 120x120 or 240x240px size. Ctrlr will do the scaling work by defining 60 width x 60 height.

**Reply by dervur** [reply]
I had no idea this existed lol could've saved me some work...

**Reply by damiensellier** [reply]
Many pro studios developping plugins still use this :) it's a great tool. Maybe there's something more modern because it's about 15 years old.

**Comment by dobo365** [comment]
Yep. Jknobman is just great!

**Reply by dervur** [reply]
I will try it out on my next panels for sure!
